src/software/geom/voronoi_util.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include "software/geom/point.h"
+#include <boost/polygon/voronoi.hpp>
+#include "software/geom/rectangle.h"
+#include "software/geom/circle.h"
+
+using boost::polygon::voronoi_builder;
+using boost::polygon::voronoi_diagram;
+using boost::polygon::voronoi_vertex;
+
+/**
+ * Finds the points of intersection between the edges of a voronoi diagram and a bounding rectangle
+ *
+ * @param vd The voronoi diagram
+ * @param rec The bounding rectangle
+ * @param points The points used to create the Voronoi diagram.
```

Comment:
Hmm, seems like we should be able to get these points from the delaunay triangulation of the given voronoi diagram?

Reply:
Also I think here (and below), there are assumptions made about the ordering of the points that may not be correct (ie. that they are in the same order as when they were used to construct the voronoi diagram)

Reply:
Why does it matter the order that the points are in?

Reply:
We're computing indexes, and them using them to index into `points`. If you reorder points before calling this function that would be an error right?

Reply:
oh yes I see the concern. I will look into it